src/bicycle/model.py

```diff
@@ -21,7 +67,45 @@ def init_weights(m):
 
 
 class Encoder(nn.Module):
-    def __init__(self, x_dim: int, z_dim: int, n_cond: int, act_fn: object = nn.GELU, hid_dim: int = 500):
+    """
+    Encoder module for a Variational Autoencoder (VAE) or similar architecture.
+
+    This class encodes input data into a latent space representation, producing
+    both the mean (`mu`) and variance (`variance`) of the latent distribution.
+    Used to sample z later, using the reparameterization trick.
+
+    Attributes:
+        z_dim (int): Dimensionality of the latent space.
+        net (nn.Sequential): A neural network that processes the input data
+            and outputs the parameters of the latent distribution.
+
+    Args:
+        x_dim (int): Dimensionality of the input data.
+        z_dim (int): Dimensionality of the latent space.
+        n_cond (int): Number of conditional variables to include in the input.
+        act_fn (object, optional): Activation function to use in the network.
+            Defaults to `nn.GELU`.
+        hid_dim (int, optional): Dimensionality of the hidden layers in the network.
+            Defaults to 500.
+
+    Methods:
+        forward(x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+            Processes the input tensor `x` and returns the mean (`mu`) and
+            variance (`variance`) of the latent distribution.
+
+    Example:
```

Comment:
I think this example can be removed, since it does not provide much information...